modules/planning/tasks/deciders/open_space_fallback_decider.cc

```diff
@@ -22,13 +22,125 @@
 
 namespace apollo {
 namespace planning {
-
 using apollo::common::Status;
+using apollo::common::math::Box2d;
+using apollo::common::math::Vec2d;
+using apollo::common::TrajectoryPoint;
+using apollo::planning::TrajGearPair;
 
 OpenSpaceFallbackDecider::OpenSpaceFallbackDecider(const TaskConfig& config)
     : Decider(config) {}
 
-Status OpenSpaceFallbackDecider::Process(Frame* frame) { return Status::OK(); }
+Status OpenSpaceFallbackDecider::Process(Frame* frame) {
+    BuildPredictedEnvironment(frame->obstacles());
+
+    if (!IsCollisionFreeTrajectory(
+      frame->open_space_info().chosen_paritioned_trajectory())) {
+      // change gflag
+      frame_->mutable_open_space_info()->set_fallback_flag(true);
+
+      // generate fallback trajectory base on current partition trajectory
+      // vehicle speed is decreased to zero inside safety distance
+      frame_->mutable_open_space_info()->set_fallback_trajectory(
+        frame->open_space_info().chosen_paritioned_trajectory());
+      double stop_distance = std::max(0.0,
+          collision_distance_ - config_.open_space_fallback_decider_config().
+              open_space_fall_back_stop_safety_gap());
+      if (stop_distance > 0.0) {
+        // the accelerate = v0^2 / (2*s), where s is slowing down distance
+        double accelerate =
+            (frame_->vehicle_state().linear_velocity() *
+             frame_->vehicle_state().linear_velocity()) /
+             2.0 / stop_distance;
+        double current_v = frame_->vehicle_state().linear_velocity();
+        size_t temp_horizon = frame_->open_space_info().
+            chosen_paritioned_trajectory().first.NumOfPoints();
+        for (size_t i = 0; i < temp_horizon; ++i) {
+          double next_v = std::max(0.0, current_v -
+              accelerate * frame_->open_space_info().
+                  chosen_paritioned_trajectory().first.
+                  TrajectoryPointAt(i).relative_time());
+          frame_->mutable_open_space_info()->
+            mutable_chosen_paritioned_trajectory()->
+            first[i].set_v(next_v);
+          frame_->mutable_open_space_info()->
+            mutable_chosen_paritioned_trajectory()->
+            first[i].set_a(-accelerate);
+          current_v = next_v;
+        }
+      } else {
+        size_t temp_horizon = frame_->open_space_info().
```

Comment:
In the "else" logic, should you set the fallback_flag back to "false" ?

Reply:
> In the "else" logic, should you set the fallback_flag back to "false" ?

The logic is if the vehicle has enough distance to the obstacle, it decrease and stop following the previous trajectory, otherwise, stop at the current point.

Reply:
> In the "else" logic, should you set the fallback_flag back to "false" ?

fixed